src/main/java/org/jsoup/nodes/Element.java

```diff
@@ -877,6 +877,27 @@ public void tail(Node node, int depth) {
     }
 
     /**
+     * Get the (unencoded) text of all children of this element, including any newlines and spaces present in the original.
+     *
+     * @return text
+     */
+    public String getWholeText(){
+        final StringBuilder accum = new StringBuilder();
+        new NodeTraversor(new NodeVisitor() {
+            public void head(Node node, int depth) {
+                if (node instanceof TextNode) {
+                    TextNode textNode = (TextNode) node;
+                    accum.append(textNode.getWholeText());
+                }
+            }
+
+            public void tail(Node node, int depth) {
+            }
+        }).traverse(this);
+        return accum.toString().trim();
```

Comment:
why `trim()`? if we use `<p>     Hello\nthere     </p>`, wouldn't I want everything inside `<p>` that is not an html node?